### PR TITLE
[master] Purge MPT roots for normal node

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description
Add purging of historical data for normal node (non-lookup)

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libData/DataStructures/TraceableDB.h [hunk 1]
```diff
@@ -110,9 +116,22 @@ class TraceableDB : public dev::OverlayDB {
  public:
   void DetachedExecutePurge() {
     LOG_MARKER();
-    auto detached_func = [this]() { return ExecutePurge(0, true); };
+    auto detached_func = [this]() -> void {
+      if (!m_purgeRunning) {
+        m_purgeRunning = true;
```
Comment (sandipbhoir) [inline src/libData/DataStructures/TraceableDB.h:121]: set m_stopSignal to false otherwise ExecutePurge might stop immediately

**Review by sandipbhoir — CHANGES_REQUESTED** [review]

File: src/libData/DataStructures/TraceableDB.h (hunk 1 above)
Comment (KaustubhShamshery) [reply src/libData/DataStructures/TraceableDB.h:121]: It is false by default and also set false after every function, it won't be true if the thread isn't running.

**Review by sandipbhoir — APPROVED** [review]

File: src/libZilliqa/Zilliqa.cpp [hunk 2]
```diff
@@ -236,6 +236,7 @@ Zilliqa::Zilliqa(const PairOfKey& key, const Peer& peer, SyncType syncType,
     if (!AccountStore::GetInstance().RefreshDB()) {
       LOG_GENERAL(WARNING, "AccountStore::RefreshDB failed");
     }
+    AccountStore::GetInstance().PurgeUnnecessary();
```
Comment (chetan-zilliqa) [inline src/libZilliqa/Zilliqa.cpp:239]: Is this function meant to be called for `seedpubs or seedprvs or seednodes`?

**Review by chetan-zilliqa — APPROVED** [review]

File: src/libZilliqa/Zilliqa.cpp (hunk 2 above)
Comment (KaustubhShamshery) [reply src/libZilliqa/Zilliqa.cpp:239]: No

**Review by sandipbhoir — APPROVED** [review]

**Review by chetan-zilliqa — APPROVED** [review]

**Pull request merged**